### PR TITLE
EditStreamCard: Better choose when to disable stream-access options

### PR DESCRIPTION
**Pull request by chrisbobbe**
See Greg's comment on [CZO](https://chat.zulip.org/#narrow/stream/378-api-design/topic/.22Stream.20administrator.22.20for.20change.20stream.20type/near/1390982):
> For the behavior here, I think one description of a good behavior would be that if you want to change a stream's access policy, you need to both
> (a) have permission to edit the stream at all (which currently means you're an org admin/owner), and
> (b) have permission to create streams with that access policy (according to `realm_create_*_stream_policy`.)
> 
> IIUC, that's the behavior we currently do have in the web UI and enforced by the server.

So, implement that behavior in the mobile UI.

Also disable the public and private options when creating a stream, since that hadn't been done yet.

### CODE REVIEW

File: src/common/SelectableOptionRow.js [hunk 1]
```diff
@@ -120,8 +123,21 @@ export default function SelectableOptionRow<TItemKey: string | number>(
...
+              url: learnMoreButton.url,
+              text: (() => {
+                const { text } = learnMoreButton;
+                switch (typeof text) {
+                  case 'undefined':
+                    return undefined;
+                  case 'string':
+                    return _(text);
+                  default:
+                    return _(text.text, text.values);
```
Comment (gnprice) [inline src/common/SelectableOptionRow.js:136]: This feels like something best abstracted into a central place, with an appropriate name. I see there's a couple other examples of it above, too.

It looks like we currently don't have a handy helper for doing this unpacking of LocalizableText; mainly we just do it in the implementation of ZulipTextIntl. I'll see about doing that as a followup PR.

**Review by gnprice — COMMENTED** [review]
Thanks @chrisbobbe! This logic looks good.

The duplication in the various disabledIfNotInitialValue expressions is pretty voluminous, though, and I think it'd be better to not duplicate -- otherwise, it's the sort of code where it'd be very easy for the different copies to diverge accidentally, with e.g. a bug being fixed in one place but not the other.

I'll push an additional commit to dedupe those. Take a look, and please go ahead and merge if it looks good to you.

File: src/common/SelectableOptionRow.js (hunk 1 above)
Comment (gnprice) [reply src/common/SelectableOptionRow.js:136]: #5429

**Comment by chrisbobbe** [comment]
Thanks for the review! Revision pushed.

**Comment by chrisbobbe** [comment]
> Take a look, and please go ahead and merge if it looks good to you.

Ah, right. Done.

**Pull request merged**